src/apply.c

```diff
@@ -211,19 +210,34 @@ static int apply_hunks(
 	git_buf *out,
 	const char *source,
 	size_t source_len,
-	git_patch *patch)
+	git_patch *patch,
+	git_diff_hunk_cb hunk_cb,
+	void *payload)
 {
 	git_patch_hunk *hunk;
 	git_diff_line *line;
 	patch_image image;
-	size_t i;
+	size_t i, skipped_lines = 0;
 	int error = 0;
 
 	if ((error = patch_image_init_fromstr(&image, source, source_len)) < 0)
 		goto done;
 
 	git_array_foreach(patch->hunks, i, hunk) {
-		if ((error = apply_hunk(&image, patch, hunk)) < 0)
+		if (hunk_cb) {
+			/* abort on <0 */
+			if ((error = hunk_cb(patch->delta, &hunk->hunk, payload)) < 0)
+				goto done;
+
+			/* skip on >0 */
+			if (error > 0) {
```

Comment:
I wonder if `> 0` is a bit broad, is that necessary?  I wonder if there may be other non-error return codes to switch on in the future.  Maybe just look for `1`...?

And if so, it might make sense to define a value here - `GIT_PATCH_SKIP`?